src/NuGet.Core/NuGet.LibraryModel/LibraryIncludeFlagUtils.cs

```diff
@@ -58,7 +66,7 @@ public static LibraryIncludeFlags GetFlags(IEnumerable<string> flags)
                         result |= LibraryIncludeFlags.Analyzers;
                         break;
                     case "buildtransitive":
-                        result |= LibraryIncludeFlags.BuildTransitive | LibraryIncludeFlags.Build;
+                        result |= includeBuildWithBuildTransitive ? LibraryIncludeFlags.BuildTransitive | LibraryIncludeFlags.Build : LibraryIncludeFlags.BuildTransitive;
```

Comment:
This is needed because the assets file is written with `BuildTransitive` but when its read for Pack, the value ends up being `Build | BuildTransitive`

Reply:
I think adding an integration test for it would be a good idea. I tried to test locally, but I struggled to find the scenario in which this change has an effect.